backend/users/views.py

```diff
@@ -277,27 +274,18 @@ class HighLevelAgencyAPI(APIView):
     permission_classes = [IsAuthenticated]
 
     def post(self, request):
-        try:
-            data = request.data
-            high_level_instance = HighLevelModel.objects.update_or_create(
-                user_id=request.user.id,
-                defaults=data
-            )
-            return Response(
+        data = request.data
+        HighLevelModel.objects.update_or_create(
+            user_id=request.user.id,
+            defaults=data
+        )
+        agency_api_key = request.data['agency_api_key']
+        data = get_celery_task(agency_api_key)
+        LocationModel.objects.bulk_create(data)
```

Comment:
So this will be done inside celery task, As there is no point of making database query here. Also let's do this by update_or_create (As you said yesterday) as there will be update functionality for it as well.